jobs/src/main/scala/dcos/metronome/model/CronSpec.scala

```diff
@@ -70,3 +77,44 @@ object CronSpec {
     }
   }
 }
+
+object CronSpecValidation {
+  val validDayOfMonth = "Day of the month must exist in the provided month and year (e.g. February have only < 29 days so running cron on 30.2. is invalid)"
+}
+
+/**
+  * In the existing validation logic inside cron-utils there is not validation that the given day exist in a month.
+  * This validator covers that use case disallowing schedules like 0 0 31 2 * (run on 31.2.)
+  */
+class CronDaysInMonthValidation extends CronConstraint(CronSpecValidation.validDayOfMonth) {
+  def daysExistInAMonths(days: Seq[Int], months: Seq[Int]): Boolean = {
+    months.exists(m => days.exists(d => dayExistInAMonth(d, m)))
+  }
+  val maxDaysOfMonth = Array(31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31)
+  def dayExistInAMonth(day: Int, month: Int): Boolean = {
+    day < maxDaysOfMonth(month - 1)
```

Comment:
`day <= maxDaysOfMonth(month - 1)` ?

Reply:
oh god, such a stupid mistake. Thanks :)!!! I refactored this thing so many times that it just got removed at one time :D